remix-debug/src/debugger/debugger.js

```diff
@@ -72,31 +72,41 @@ Debugger.prototype.debug = function (blockNumber, txNumber, tx, loadingCb) {
   const self = this
   let web3 = this.debugger.web3
 
-  if (this.debugger.traceManager.isLoading) {
-    return
-  }
+  return new Promise((resolve, reject) => {
+    if (this.debugger.traceManager.isLoading) {
+      return resolve()
+    }
 
-  if (tx) {
-    if (!tx.to) {
-      tx.to = traceHelper.contractCreationToken('0')
+    if (tx) {
+      if (!tx.to) {
+        tx.to = traceHelper.contractCreationToken('0')
+      }
+      self.debugTx(tx, loadingCb)
+      return resolve()
     }
-    return self.debugTx(tx, loadingCb)
-  }
 
-  try {
-    if (txNumber.indexOf('0x') !== -1) {
-      return web3.eth.getTransaction(txNumber, function (_error, result) {
+    try {
+      if (txNumber.indexOf('0x') !== -1) {
+        return web3.eth.getTransaction(txNumber, function (_error, result) {
+          if (_error) return reject(_error)
+          if (!result) return reject('cannot find transaction ' + txNumber)
+          let tx = result
```

Comment:
Why not calling the result `tx` from the start ? 
```javascript
web3.eth.getTransaction(txNumber, (_error, tx) => {
  if (_error) return reject(_error)
  if (!result) return reject('cannot find transaction ' + txNumber)
  this.debugTx(tx, loadingCb)
  resolve()
})
```

Reply:
yeah well I just make it returning a promise without thinking to much on rewriting it.
Let me have a look on it